test/micro/org/openjdk/bench/jdk/classfile/ConstantPoolBuildingClassEntry.java

```diff
@@ -0,0 +1,134 @@
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package org.openjdk.bench.jdk.classfile;
+
+import java.lang.classfile.constantpool.ConstantPoolBuilder;
+import java.lang.constant.ClassDesc;
+import java.util.List;
+import java.util.Random;
+import java.util.concurrent.TimeUnit;
+
+import jdk.internal.classfile.impl.Util;
+import org.openjdk.jmh.annotations.*;
+import org.openjdk.jmh.infra.Blackhole;
+
+import static java.lang.constant.ConstantDescs.*;
+
+import static org.openjdk.bench.jdk.classfile.TestConstants.*;
+
+/**
+ * Tests constant pool builder lookup performance for existing entries.
+ */
+@Warmup(iterations = 3)
+@Measurement(iterations = 5)
+@OutputTimeUnit(TimeUnit.MILLISECONDS)
+@BenchmarkMode(Mode.Throughput)
+@Fork(value = 1, jvmArgsAppend = {
+        "--enable-preview",
+        "--add-exports", "java.base/jdk.internal.classfile.impl=ALL-UNNAMED"})
+@State(Scope.Benchmark)
+public class ConstantPoolBuildingClassEntry {
+    // JDK-8338546
+    static final int RUNS = 0xff; // Q: Should we measure consecutive runs at all?
+    ConstantPoolBuilder builder;
+    List<ClassDesc> classDescs;
+    List<ClassDesc> nonIdenticalClassDescs;
+    List<String> internalNames;
+    Random random;
+    int size;
+
+    @Setup(Level.Iteration)
+    public void setup() {
+        builder = ConstantPoolBuilder.of();
+        classDescs = List.of(
+                CD_Byte, CD_Object, CD_Long.arrayType(), CD_String, CD_String, CD_Object, CD_Short,
+                CD_MethodHandle, CD_MethodHandle, CD_Object, CD_Character, CD_List, CD_ArrayList,
+                CD_List, CD_Set, CD_Integer, CD_Object.arrayType(), CD_Enum, CD_Object, CD_MethodHandles_Lookup,
+                CD_Long, CD_Set, CD_Object, CD_Character, CD_Integer, CD_System, CD_String, CD_String,
+                CD_CallSite, CD_Collection, CD_List, CD_Collection, CD_String
+        );
+        size = classDescs.size();
+        nonIdenticalClassDescs = classDescs.stream().map(cd -> {
+            var ret = ClassDesc.ofDescriptor(cd.descriptorString());
+            ret.hashCode(); // pre-compute hash code
+            return ret;
+        }).toList();
+        internalNames = classDescs.stream().map(cd -> {
+            // also sets up builder
+            var ce = builder.classEntry(cd); // pre-computes hash code for cd
+            return ce.name().stringValue(); // pre-computes hash code for stringValue
+        }).toList();
+        random = new Random(-555278801022917158L);
+    }
+
+    /**
+     * Looking up with identical ClassDesc objects. Happens in bytecode generators reusing
+     * constant CD_Xxx.
+     */
+    @Benchmark
+    public void identicalLookup(Blackhole bh) {
+        for (int i = 0; i < RUNS; i++) {
+            int n = random.nextInt(size);
```

Comment:
Why the randomization here? `Random` can be pretty slow in and off itself, skewing results, and scales poorly. Prefer `ThreadLocalRandom`. `classDescs` look pretty random to begin with, so why not just loop over it once per measurement?